src/query/RelayQuery.js

```diff
@@ -214,6 +218,36 @@ class RelayQueryNode {
     return children;
   }
 
+  shouldBeIncluded(): ?boolean {
+    var directives = this.getDirectives();
+
+    var skipNode = directives.find(
+      directive => directive.name === SKIP
+    );
+    if (skipNode) {
+      var skipIf = skipNode.arguments.find(
+        arg => arg.name === IF
+      );
+      if (skipIf) {
+        return !skipIf.value;
+      }
+    }
+
+    var includeNode = directives.find(
+      directive => directive.name === INCLUDE
+    );
+    if (includeNode) {
+      var includeIf = includeNode.arguments.find(
+        arg => arg.name === IF
+      );
+      if (includeIf) {
+        return Boolean(includeIf.value);
+      }
+    }
+
+    return true
```

Comment:
Good idea for the iteration, I will add that in tonight. I dont think `isSkipped || !isIncluded` will work though. Since `skip` should always have priority over `include`, something like `False || !False` would result in the field being skipped even though skip was `False`!